lib/generic_stacktrace.c

```diff
@@ -102,23 +102,35 @@ sr_stacktrace_parse(enum sr_report_type type, const char *input, char **error_me
 }
 
 struct sr_stacktrace *
-sr_stacktrace_from_json(enum sr_report_type type, struct sr_json_value *root, char **error_message)
+sr_stacktrace_from_json(enum sr_report_type type, json_object *root, char **error_message)
 {
     return DISPATCH(dtable, type, from_json)(root, error_message);
 }
 
 struct sr_stacktrace *
 sr_stacktrace_from_json_text(enum sr_report_type type, const char *input, char **error_message)
 {
-    struct sr_json_value *json_root = sr_json_parse(input, error_message);
+    enum json_tokener_error error;
+    json_object *json_root = json_tokener_parse_verbose(input, &error);
 
     if (!json_root)
+    {
+        if (NULL != error_message)
+        {
+            const char *description;
+
+            description = json_tokener_error_desc(error);
```

Comment:
Why not squash the declaration and assignment?

Reply:
That’s just poor form, IMO, mixing code and declarations.

Reply:
OK, that is your prerogative.